### ISSUE TITLE
NLog Database target issue in 4.6.6

### ISSUE THREAD

**Issue by Rajkumar28269** [NLog/NLog#3528]
**NLog 4.6.6**

**Platform**: .Net 4.5
**Current NLog config** (xml or C#, if relevant)

<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
      autoReload="true"
      throwExceptions="true"
      internalLogLevel="Debug" internalLogFile="c:\temp\nlog-internal.log">

  <targets>
    <target name="logfile" xsi:type="File"
                layout="___________________ ${level} (${longdate}) ______________________
                      ${newline}
                      Call Site: ${callsite}${newline}
                      Exception Type: ${exception:format=Type}${newline}
                      Exception Message: ${exception:format=Message}${newline}
                      Stack Trace: ${exception:format=StackTrace}${newline}
                      Additional Info: ${message}${newline}"
                fileName="${basedir}/${level}.log"
                maxArchiveFiles="4"
                archiveAboveSize="10240"
                archiveEvery="Day" />

    <target xsi:type="Database" name="logdb">
    </target>
  </targets>
  <rules>
    <logger name="*" levels="Info" writeTo="logdb" />
    <logger name="*" levels="Error,Trace,Debug" writeTo="logfile" />
  </rules>
</nlog>

- What is the current result?
System.TypeInitializationException was unhandled
Message: An unhandled exception of type 'System.TypeInitializationException' occurred in mscorlib.dll
Additional information: The type initializer for 'NLog_Console.Program' threw an exception.

- What is the expected result?
It should allow me to use database target. when I comment database target it works fine with file target.

- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
Yes
- Are there any workarounds?
no

**Comment by 304NotModified** [comment]
Please check this https://stackoverflow.com/questions/34994470/better-typeinitializationexception-innerexception-is-also-null

**Comment by 304NotModified** [comment]
Tl Dr. In your program class there is a static logger field and the logger throws an error.

**Comment by Rajkumar28269** [comment]
Thank you got it now working...

**Comment by 304NotModified** [comment]
I'm glad is has been resolved 👍
